### ISSUE TITLE
Bug: bitarray is hashable on Python 2, causes KeyError

### ISSUE THREAD

**Issue by dargueta** [ilanschnell/bitarray#74]
Mutable objects shouldn't be able to be used as keys, as they can cause collisions and unexpected behavior. I haven't dug into why yet but on Python 2 bitarrays *are* hashable, allowing programmers to use them as dictionary keys.

To reproduce:

```py
import bitarray

array_one = bitarray.bitarray([True, False])
array_two = bitarray.bitarray([False, False])

dct = {
    array_one: "one",
    array_two: "two",
}

array_three = bitarray.bitarray([True, False])
array_three == array_one  # Returns True

dct[array_three]   # Should return "one", throws KeyError instead
```

**Comment by ilanschnell** [comment]
Thanks!  You are correct, use like lists, bitarrays should always be unhashable.  I just tried it on Python 3.7, and indeed `hash(bitarray())` gives a `TypeError`, whereas on Python 2.7 it does not.  I'll have to look into why this is the case.

**Comment by ilanschnell** [comment]
I just fixed the issue in 273f1f731f16e212dffc8cd6b0830036fa790fbe.  This fix will be part of the upcoming 1.1.0 release.

**Comment by ilanschnell** [comment]
FYI, I've added a `frozenbitarray` object in master, which will also be part of the next release, so your example will work fine, if this is used instead of the regular `bitarray`.
